dockerfiles/build-extension/Dockerfile_alpine

```diff
@@ -0,0 +1,45 @@
+FROM alpine:3.11 as build
+
+WORKDIR /php-src
+ADD .php-src .
+
+ARG php_version
+
+# We are only downloading from main repository
+RUN sed -i '/main$/!d' /etc/apk/repositories
+
+RUN apk add --no-cache \
+    alpine-sdk \
+    autoconf \
+    bash \
+    bison \
+    git \
+    curl-dev \
+    re2c \
+    libxml2-dev \
+    libexecinfo-dev
+
+RUN git checkout PHP-${php_version}
+RUN ./buildconf
+RUN ./configure --prefix=/php-install --without-sqlite3 --without-pdo-sqlite --with-curl
```

Comment:
@SammyK @morrisonlevi any specific flag you would set in the machine that we use to build alpine images?

Reply:
I wouldn't use `--enable-debug` because that changes the API, but I would make sure that the debugging symbols are there. Currently I'm debugging a segfault on Alpine that only exists with our APK, and the PHP apk doesn't have a `-dbg` version :'(

I think it's default to have symbols and Alpine `strip`s them, but just making it explicit because it's a super annoying thing.

Reply:
Do you think this should be part of the images we use to build alpine or we should have dev images specific to alpine?

Reply:
I think we need debugging symbols everywhere, you never know when you are going to get a sigsegv so I'd much rather have them available.

Here are the flags that the alpine container I am in used:

```
--build=x86_64-linux-musl \
--with-config-file-path=/usr/local/etc/php \
--with-config-file-scan-dir=/usr/local/etc/php/conf.d \
--enable-option-checking=fatal \
--with-mhash \
--enable-ftp \
--enable-mbstring \
--enable-mysqlnd \
--with-password-argon2 \
--with-sodium=shared \
--with-pdo-sqlite=/usr \
--with-sqlite3=/usr \
--with-curl \
--with-libedit \
--with-openssl \
--with-zlib \
build_alias=x86_64-linux-musl
```
I think we probably want to have `libedit` or `readline` so we can do `php -a`. Probably want mbstring and zlib because I think composer requires them (or is it PhpUnit?).

Reply:
Is there a reason why we can't (or shouldn't) use our existing Alpine images for this? We started moving those from `docker-library` to `dd-trace-ci` starting with `datadog/dd-trace-ci:php-7.4-debug-alpine-3.11`. https://github.com/DataDog/dd-trace-ci/blob/master/php/alpine/Dockerfile

Reply:
Do the existing Alpine images have debugging symbols for PHP? If not, that alone is enough for me to vote to remake them.

Reply:
That's a good question. What was the original motivation behind those images? I wonder if it might be a little bit difficult to follow whats going on there vs having the simplest image possible (from the point of view of installed dependencies) dedicated to the build.